deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/impl/paramavg/ParameterAveragingTrainingMaster.java

```diff
@@ -160,30 +162,40 @@ public ParameterAveragingTrainingMaster(boolean saveUpdater, Integer numWorkers,
      * @param rddDataSetNumExamples Number of examples in each DataSet object in the {@code RDD<DataSet>}
      * @param batchSizePerWorker    Number of examples to use per worker per fit
      * @param averagingFrequency    Frequency (in number of minibatches) with which to average parameters
+     * @param aggregationDepth      Number of aggregation levels used in parameter aggregation
      * @param prefetchNumBatches    Number of batches to asynchronously prefetch (0: disable)
+     * @param repartition           Set if/when repartitioning should be conducted for the training data
+     * @param repartitionStrategy   Repartitioning strategy to use. See {@link RepartitionStrategy}
      * @param collectTrainingStats  If true: collect training statistics for debugging/optimization purposes
      */
     public ParameterAveragingTrainingMaster(boolean saveUpdater, Integer numWorkers, int rddDataSetNumExamples,
-                    int batchSizePerWorker, int averagingFrequency, int prefetchNumBatches, Repartition repartition,
-                    RepartitionStrategy repartitionStrategy, boolean collectTrainingStats) {
-        this(saveUpdater, numWorkers, rddDataSetNumExamples, batchSizePerWorker, averagingFrequency, prefetchNumBatches,
-                        repartition, repartitionStrategy, StorageLevel.MEMORY_ONLY_SER(), collectTrainingStats);
+                    int batchSizePerWorker, int averagingFrequency, int aggregationDepth, int prefetchNumBatches,
+                    Repartition repartition, RepartitionStrategy repartitionStrategy, boolean collectTrainingStats) {
+        this(saveUpdater, numWorkers, rddDataSetNumExamples, batchSizePerWorker, averagingFrequency, aggregationDepth,
+                prefetchNumBatches, repartition, repartitionStrategy, StorageLevel.MEMORY_ONLY_SER(), collectTrainingStats);
     }
 
     public ParameterAveragingTrainingMaster(boolean saveUpdater, Integer numWorkers, int rddDataSetNumExamples,
-                    int batchSizePerWorker, int averagingFrequency, int prefetchNumBatches, Repartition repartition,
-                    RepartitionStrategy repartitionStrategy, StorageLevel storageLevel, boolean collectTrainingStats) {
+                    int batchSizePerWorker, int averagingFrequency, int aggregationDepth, int prefetchNumBatches,
+                    Repartition repartition, RepartitionStrategy repartitionStrategy, StorageLevel storageLevel,
+                    boolean collectTrainingStats) {
         if (numWorkers <= 0)
             throw new IllegalArgumentException("Invalid number of workers: " + numWorkers + " (must be >= 1)");
         if (rddDataSetNumExamples <= 0)
             throw new IllegalArgumentException(
                             "Invalid rdd data set size: " + rddDataSetNumExamples + " (must be >= 1)");
+        if (averagingFrequency <= 0)
```

Comment:
Great initiative, but we use `[Guava's preconditions](https://github.com/google/guava/wiki/PreconditionsExplained)` to do checks of this sort.